### PR TITLE
Add own telemetry controller listening on shoot telemetry resources

### PR DESCRIPTION
**Pull request by schrodit**
**What this PR does / why we need it**:

The telemetry controller is added that wraps the ShootTelemetry controller to measure shoots via CRD's.

This is the first step to bring the testrunner logic into the TM and make it more resilient and usable

**Special notes for your reviewer**:

**Release note**:

```improvement operator
The telemetry controller is added that wraps the ShootTelemetry controller to measure shoots via CRD's.
```

### CODE REVIEW

File: cmd/telemetry-controller/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Telemetry Controller
+
+The telemetry controller is a kubernetes controller that wraps the shoot telemetry controller with a CRD.
+With the telemetry controller the measurement of shoots can be instructed using a ShootsMeasurements CRD.
+
+### Development
+```
+go run ./cmd/telemetry-controller/main.go
+  --cache-dir=/tmp/tl # specify the directory path for the measurements cache 
+  --kubeconfig=/.kube/config # optional, uses teh in cluster config if not defined
```
Comment (OlegLoewen) [inline cmd/telemetry-controller/README.md]:
```suggestion
  --kubeconfig=/.kube/config # optional, uses the in cluster config if not defined
```

File: charts/telemetry/templates/controller.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        checksum/tls-secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+      labels:
+        app: telemetry-controller
+    spec:
+{{ if .Values.controller.imagePullSecretName }}
+      imagePullSecrets:
+      - name: "{{.Values.controller.imagePullSecretName}}"
+{{end}}
+      serviceAccountName: {{ required ".Values.controller.serviceAccountName is required" .Values.controller.serviceAccountName }}
+      containers:
```
[the remark this answers is not in the document]
Comment (schrodit) [reply charts/telemetry/templates/controller.yaml]: added default limits and made them configurable in the helm chart

File: pkg/telemetry/controller/reconciler.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+func (r *telemetryReconciler) stop(st *telv1beta1.ShootsMeasurement) error {
+	if st.Status.Data == nil {
+		st.Status.Data = make([]telv1beta1.ShootMeasurementData, 0)
+	}
+	for _, shoot := range st.Spec.Shoots {
+		fig, err := r.controllerManager.StopAndAnalyze(st.Status.Controller, shoot)
+		if err != nil {
+			st.Status.Phase = telv1beta1.TelemetryPhaseError
+			st.Status.Message = fmt.Sprintf("unable to get measuremtn for %s", shoot.String())
```
Comment (OlegLoewen) [inline pkg/telemetry/controller/reconciler.go]:
```suggestion
			st.Status.Message = fmt.Sprintf("unable to get measurement for %s", shoot.String())
```

File: pkg/telemetry/manager/manager.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+	// Load the kubeconfig.
+	configObj, err := clientcmd.Load(kubeconfigRaw)
+	if err != nil {
+		return nil, err
+	} else if configObj == nil {
+		return nil, errors.New("config is nil")
+	}
+
+	// Create a kubernets client config.
```
Comment (OlegLoewen) [inline pkg/telemetry/manager/manager.go]:
```suggestion
	// Create a kubernetes client config.
```

File: pkg/testrunner/telemetry/telemetry.go [hunk 5]
```diff
@@ -148,14 +168,36 @@ func (c *Telemetry) IsStarted() bool {
...
 }
 
 // RemoveShoot removes a shoot from the telemetry watch
 func (c *Telemetry) RemoveShoot(shootKey string) {
+	c.mut.Lock()
+	defer c.mut.Unlock()
 	delete(c.shootsFilter, shootKey)
 }
 
+// HasShoot returns if a shoot is measured
```
Comment (OlegLoewen) [inline pkg/testrunner/telemetry/telemetry.go]:
```suggestion
// HasShoot returns true if a shoot is measured
```

**Pull request merged**